### DISCUSSION TITLE
Run in headless mode on a server?

### DISCUSSION THREAD

**Discussion by aayushdutt** [jwallet/spy-spotify#478] [Ideas]
I know as of now this is only supported using windows app, but having the ability to record it automatically and periodically in a  headless server would be awesome. 
Users could provide the Spotify credentials and it streams and records headlessly - potentially saving the files to GDrive/iCloud etc.
This will also help avoid keeping the computer on for a hundred hours if the playlist is large, and be updated with the latest changes.

**Comment by goguetchapuisb** [comment]
But I think Spotify can't run in headless mode since a moment. Maybe on a virtual machine or on raspbian desktop... (My idea was to put it on a raspberry pi)

**Comment by StefanDorschu** [comment]
I'm running it on a vm on a server but when disconnecting the error occurs that Spotify and spy-spotify don't use the same audio devices. But they use the same as configured.

Any ideas?

**Comment by goguetchapuisb** [comment]
I don't have test a windows server but i think you close your session and
Spotify is closed too...

Baptiste Goguet-Chapuis
06.25.13.36.47

Le mer. 1 mars 2023 à 18:14, HabelStefan ***@***.***> a
écrit :

**Reply by StefanDorschu** [reply]
No, everything is still running, Spotify is playing. The only thing which happens is that the "error" occurs.
